### PR TITLE
Use the correct OperationIdGenerator for OperationOutput

### PR DESCRIPTION
**Pull request by tasomaniac**
Use the correct OperationIdGenerator for OperationOutput

Since only used in single place, move sha256 implementation and make it private

### CODE REVIEW

**Comment by tasomaniac** [comment]
@Laimiux After your contribution, a custom hashing (operation id) support was added in #1849. Can you have a look if this looks correct?

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationIdGenerator.kt [hunk 1]
```diff
@@ -22,11 +25,18 @@ interface OperationIdGenerator {
    */
   val version: String
 
-  class Sha256: OperationIdGenerator {
+  class Sha256 : OperationIdGenerator {
     override fun apply(operationDocument: String, operationFilepath: String): String {
       return operationDocument.sha256()
     }
 
     override val version = "1.0"
```
Comment (tasomaniac) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationIdGenerator.kt]: Should we have a name like `Sha256-1.0`?

I feel like `1.0` is really generic and another algoritm should not really have the same version.

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/operationoutput/OperationOutputWriter.kt [hunk 2]
```diff
@@ -1,27 +1,24 @@
...
+import com.apollographql.apollo.compiler.OperationIdGenerator
 import com.apollographql.apollo.compiler.ir.CodeGenerationIR
 import com.apollographql.apollo.compiler.ir.Operation
-import com.apollographql.apollo.compiler.sha256
 import com.apollographql.apollo.internal.QueryDocumentMinifier
 import java.io.File
-import com.squareup.moshi.Moshi
-import com.squareup.moshi.Types
 
-internal class OperationOutputWriter(val packageNameProvider: PackageNameProvider) {
```
Comment (tasomaniac) [inline apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/operationoutput/OperationOutputWriter.kt]: `packageNameProvider` was never used.

**Comment by Laimiux** [comment]
Looks good to me!

File: apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationIdGenerator.kt (hunk 1 above)
Comment (martinbonnin) [reply apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationIdGenerator.kt]: +1. We should definitely encourage unique version names.

Comment (tasomaniac) [reply apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/OperationIdGenerator.kt]: Just pushed a commit. Plz have a look.

**Pull request merged**